packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -145,66 +168,407 @@ String getGradleVersionForAndroidPlugin(Directory directory, Logger logger) {
+  return false;
+}
+
+
+
 /// Returns the Gradle version that is required by the given Android Gradle plugin version
 /// by picking the largest compatible version from
 /// https://developer.android.com/studio/releases/gradle-plugin#updating-gradle
-@visibleForTesting
 String getGradleVersionFor(String androidPluginVersion) {
-  if (_isWithinVersionRange(androidPluginVersion, min: '1.0.0', max: '1.1.3')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '1.0.0', max: '1.1.3')) {
     return '2.3';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '1.2.0', max: '1.3.1')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '1.2.0', max: '1.3.1')) {
     return '2.9';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '1.5.0', max: '1.5.0')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '1.5.0', max: '1.5.0')) {
     return '2.2.1';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '2.0.0', max: '2.1.2')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '2.0.0', max: '2.1.2')) {
     return '2.13';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '2.1.3', max: '2.2.3')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '2.1.3', max: '2.2.3')) {
     return '2.14.1';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '2.3.0', max: '2.9.9')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '2.3.0', max: '2.9.9')) {
     return '3.3';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '3.0.0', max: '3.0.9')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '3.0.0', max: '3.0.9')) {
     return '4.1';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '3.1.0', max: '3.1.9')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '3.1.0', max: '3.1.9')) {
     return '4.4';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '3.2.0', max: '3.2.1')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '3.2.0', max: '3.2.1')) {
     return '4.6';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '3.3.0', max: '3.3.2')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '3.3.0', max: '3.3.2')) {
     return '4.10.2';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '3.4.0', max: '3.5.0')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '3.4.0', max: '3.5.0')) {
     return '5.6.2';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '4.0.0', max: '4.1.0')) {
+  if (isWithinVersionRange(androidPluginVersion, min: '4.0.0', max: '4.1.0')) {
     return '6.7';
   }
-  if (_isWithinVersionRange(androidPluginVersion, min: '7.0', max: '7.5')) {
+  // 7.5 is a made up value to include everything through 7.4.*
+  if (isWithinVersionRange(androidPluginVersion, min: '7.0', max: '7.5')) {
     return '7.5';
   }
+  if (isWithinVersionRange(androidPluginVersion, min: '7.5', max: maxKnownAgpVersion)) {
+    return '8.0';
+  }
```

Comment:
Similar optional nit here; a map of version ranges and trivial loop could replace all the duplicate calls.

Reply:
I like that idea. Similar to how I did the unit tests. Let me work on that today.

Reply:
I think this worked out very nicely!